src/graph/entities/graph_entity.c

```diff
@@ -78,7 +87,13 @@ SIValue *GraphEntity_GetProperty(const GraphEntity *e, Attribute_ID attr_id) {
 
 // Updates existing property value.
 void GraphEntity_SetProperty(const GraphEntity *e, Attribute_ID attr_id, SIValue value) {
-	assert(e);
+	ASSERT(e);
+
+	// Emit an error and return if we're trying to add an invalid type.
+	if(!(SI_TYPE(value) & SI_PRIMITIVES)) {
+		QueryCtx_SetError("Property values can only be of primitive types.");
+		return;
+	}
```

Comment:
Same verification as in `GraphEntity_AddProperty` consider moving into a static inline function.